projects/plugins/social/src/class-meta-tags.php

```diff
@@ -0,0 +1,223 @@
+	 */
+	public function get_active_plugins() {
+		$active_plugins = (array) get_option( 'active_plugins', array() );
+
+		if ( is_multisite() ) {
+			// Due to legacy code, active_sitewide_plugins stores them in the keys,
+			// whereas active_plugins stores them in the values.
+			$network_plugins = array_keys( get_site_option( 'active_sitewide_plugins', array() ) );
+			if ( $network_plugins ) {
+				$active_plugins = array_merge( $active_plugins, $network_plugins );
+			}
+		}
+
+		sort( $active_plugins );
+
+		return array_unique( $active_plugins );
+	}
+
+	/**
+	 * Check if meta tags should be rendered.
+	 *
+	 * @return bool True if meta tags should be rendered.
+	 */
+	public function should_render_meta_tags() {
+		if ( ! empty( array_intersect( $this->get_active_plugins(), $this->open_graph_conflicting_plugins ) ) ) {
+			return false;
+		}
+
+		if ( ! is_singular() ) {
+			return false;
+		}
+
+		if ( ! apply_filters( 'jetpack_enable_open_graph', true ) ) {
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Check if Twitter Cards tags should be rendered.
+	 *
+	 * @return bool True if Twitter Cards tags should be rendered.
+	 */
+	public function should_render_twitter_cards_tags() {
+		return empty( array_intersect( $this->get_active_plugins(), $this->twitter_cards_conflicting_plugins ) );
+	}
+
+	/**
+	 * Get the featured image for a post.
+	 *
+	 * @param int $post_id The post ID. Optional. Defaults to global $post.
+	 * @param int $width   The minimum width of the image. Optional. Defaults to 200.
+	 * @param int $height  The minimum height of the image. Optional. Defaults to 200.
+	 * @return array The featured image and dimensions. Empty array if no image is found.
+	 */
+	public function get_featured_image( $post_id = 0, $width = 200, $height = 200 ) {
+		$post = get_post( $post_id );
+
+		if ( ! has_post_thumbnail( $post ) ) {
+			return array();
+		}
+
+		if ( ! empty( $post->post_password ) ) {
+			return array();
+		}
+
+		$thumb = get_post_thumbnail_id( $post );
+		$meta  = wp_get_attachment_metadata( $thumb );
+
+		// Must be larger than requested minimums.
+		if ( ! isset( $meta['width'] ) || $meta['width'] < $width ) {
+			return array();
+		}
+
+		if ( ! isset( $meta['height'] ) || $meta['height'] < $height ) {
+			return array();
+		}
+
+		$img_src = wp_get_attachment_image_src( $thumb, array( 1200, 1200 ) );
+
+		if ( empty( $img_src ) ) {
+			return array();
+		}
+
+		return array(
+			'url'    => $img_src[0],
+			'width'  => $img_src[1],
+			'height' => $img_src[2],
+		);
+	}
+
+	/**
+	 * Render meta tags in head.
+	 */
+	public function render_tags() {
+		if ( ! $this->should_render_meta_tags() ) {
+			return;
+		}
+
+		$image = $this->get_featured_image();
+
+		if ( empty( $image ) ) {
+			return;
+		}
+
+		$tags = array(
+			'og:image'        => $image['url'],
+			'og:image:width'  => $image['width'],
+			'og:image:height' => $image['height'],
+		);
+
+		if ( $this->should_render_twitter_cards_tags() ) {
```

Comment:
Bit of a nitpick, but can we break this into an function? 

```
$tags = $this->append_service_specific_tags( $image );
```

and have 

```
function append_service_specific_tags( $tags, $image ) {

    if (  $this->should_render_twitter_cards_tags() ) {
         $tags = array_merge(
				$tags,
				array(
					'twitter:image' => $image['url'],
					'twitter:card'  => 'summary_large_image',
				)
			);
     }
     return $tags; 

 }
 
 
 This way, if we needed to add any service-specific tags in the future, we can do that in the `append_service_specific_tags` method. 
 
```

Reply:
Let's leave it for now. This whole class is a stop gap. We can refine things in the package